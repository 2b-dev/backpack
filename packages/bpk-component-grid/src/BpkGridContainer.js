import React, { PropTypes } from 'react';

import './bpk-grid.scss';

const BpkGridContainer = (props) => {
  const { children, className, debug, fullWidth, ...rest } = props;
  const classNames = ['bpk-grid__container'];

  if (debug) { classNames.push('bpk-grid__container--debug'); }
  if (fullWidth) { classNames.push('bpk-grid__container--full-width'); }
  if (className) { classNames.push(className); }

  return <div className={classNames.join(' ')} {...rest}>{children}</div>;
};

BpkGridContainer.propTypes = {
  children: PropTypes.node.isRequired,
  debug: PropTypes.bool,
  fullWidth: PropTypes.bool,
  className: PropTypes.string,
};

BpkGridContainer.defaultProps = {
  debug: false,
  fullWidth: false,
  className: null,
};

export default BpkGridContainer;
