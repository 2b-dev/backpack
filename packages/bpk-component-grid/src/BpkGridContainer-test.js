import React from 'react';
import renderer from 'react/lib/ReactTestRenderer';
import BpkGridContainer from './BpkGridContainer';

describe('BpkGridContainer', () => {
  it('should render correctly', () => {
    const tree = renderer.create(
      <BpkGridContainer>Contents</BpkGridContainer>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly with "padded" attribute equal to false', () => {
    const tree = renderer.create(
      <BpkGridContainer padded={false}>Contents</BpkGridContainer>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly with "debug" attribute', () => {
    const tree = renderer.create(
      <BpkGridContainer debug>Contents</BpkGridContainer>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

