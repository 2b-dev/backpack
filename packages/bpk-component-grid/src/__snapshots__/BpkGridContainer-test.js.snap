exports[`BpkGridContainer should render correctly 1`] = `
<div
  className="bpk-grid__container bpk-grid__container--padded">
  Contents
</div>
`;

exports[`BpkGridContainer should render correctly with "debug" attribute 1`] = `
<div
  className="bpk-grid__container bpk-grid__container--padded bpk-grid__container--debug">
  Contents
</div>
`;

exports[`BpkGridContainer should render correctly with "padded" attribute equal to false 1`] = `
<div
  className="bpk-grid__container">
  Contents
</div>
`;
